pkg/image/registry/imagestreamimport/rest.go

```diff
@@ -305,9 +305,11 @@ func checkImportFailure(status api.ImageImportStatus, stream *api.ImageStream, t
 		if len(status.Tag) > 0 {
 			tag = status.Tag
 		} else if status.Image != nil {
-			if ref, err := api.ParseDockerImageReference(status.Image.DockerImageReference); err == nil {
-				tag = ref.Tag
+			ref, err := api.ParseDockerImageReference(status.Image.DockerImageReference)
+			if err != nil || ref.Tag == "" {
+				return false
```

Comment:
Thanks for review! @liggitt 
`api.HasTagCondition` and `api.SetTagCondition` are meaningless, if `tag` is an empty string.
So we have to ensure that `tag` is a non empty string.